mockgcp/common/httptogrpc/dynamicgrpcgateway.go

```diff
@@ -34,7 +34,7 @@ import (
 // Mux is the primary interface for mapping HTTP requests to gRPC method calls.
 type Mux interface {
 	http.Handler
-	AddService(client any)
+	AddService(client any, opt ...ServiceOption)
```

Comment:
This change to `AddService` breaks the `Mux` interface for any external implementations.
If `mockgcp` is consumed by others who implement `Mux`, this is a breaking change.
Given this seems internal to the project's test suite, it's likely fine.

Reply:
Right, I don't think httptogrpc is yet being used outside of KCC.  Also maybe this shouldn't even be an interface.